src/main/java/it/gov/pagopa/payhub/activities/dto/ingestion/CsvReadResult.java

```diff
@@ -0,0 +1,19 @@
+package it.gov.pagopa.payhub.activities.dto.ingestion;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Stream;
+
+@Data
+@AllArgsConstructor
+public class CsvReadResult<T> {
+    private final Stream<T> dataStream;
+    private final AtomicLong totalRows;
```

Comment:
I see not advantage on handling the count on this manner...
totalRows will be not completed untile the stream is not totally read...

a structure like this could let think to be able to have the totalRows without even read the stream...
let's handle the counter directly on stream reading... (ps: why don't we just return the iterator itself?